### PR TITLE
Schedule filters

### LINKED ISSUE #160: Org: Filters for Scedule
One the Org schedule page we need to add these filters

Image

accessible by the filter button

Image

### PR DESCRIPTION
**Pull request by adescoteaux1**
# Description
[Link to Ticket](insert the link to your ticket inside the parenthesis here)
Please include a summary of the changes and the related issue. Please also
include relevant motivation, context, and images!

# How Has This Been Tested?
Please describe the tests that you manually ran to verify your changes (beyond any unit/integration tests written and ran).

## Screenshots
Please provide screenshots of manual testing (Scalar, frontend pages, etc.)

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

File: frontend/apps/mobile/app/(app)/org/[id]/schedule.tsx [hunk 1]
```diff
@@ -32,11 +31,15 @@ export default function OrgScheduleScreen() {
...
-  const [selectedClass, setSelectedClass] = useState<string | null>(
-    filterClass ?? null,
-  );
-  const [filterVisible, setFilterVisible] = useState(false);
+  // Honour the filterClass URL param on first mount
+  useEffect(() => {
+    if (filterClass && !filters.class_name) {
+      setFilters({ ...filters, class_name: filterClass });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/org/[id]/schedule.tsx]: Erm

File: frontend/apps/mobile/app/(app)/org/[id]/schedule.tsx [hunk 2]
```diff
@@ -48,18 +51,70 @@ export default function OrgScheduleScreen() {
...
+    let result = occurrences;
 
-  const filteredOccurrences = useMemo(
-    () =>
-      selectedClass
-        ? occurrences.filter((o) => o.event.title === selectedClass)
-        : occurrences,
-    [occurrences, selectedClass],
-  );
+    if (filters.class_name) {
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/org/[id]/schedule.tsx]: Can we clean this up a lil?

File: frontend/apps/mobile/app/(app)/org/[id]/schedule.tsx [hunk 3]
```diff
@@ -89,9 +144,9 @@ export default function OrgScheduleScreen() {
 
   const groupOccurrencesByDate = useCallback(
     (
-      occurrences: EventOccurrence[],
+      occs: EventOccurrence[],
     ): { label: string; items: EventOccurrence[] }[] => {
-      const sorted = [...occurrences].sort(
+      const sorted = [...occs].sort(
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/org/[id]/schedule.tsx]: Wondering if we should memoize the sorting? Nbd because its wrapped in a callback but could save a lil

File: frontend/apps/mobile/app/(app)/org/[id]/schedule.tsx [hunk 4]
```diff
@@ -146,9 +201,9 @@ export default function OrgScheduleScreen() {
           {translate("org.schedule")}
         </Text>
         <TouchableOpacity
-          onPress={() => setFilterVisible(true)}
+          onPress={() => router.push(`/org/${id}/filters` as never)}
```
Comment (josh-torre) [inline frontend/apps/mobile/app/(app)/org/[id]/schedule.tsx]: remove type cast?

**Review by josh-torre — APPROVED** [review]
LGTM 🚀
